File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
  let answer = "";
  let mode = 0;

  for (let i = 0; i < code.length; i++) {
    if (mode === 0) {
      code[i] == "1" ? (mode = 1) : i % 2 === 0 && (answer += code[i]);
    } else {
      code[i] == "1" ? (mode = 0) : i % 2 !== 0 && (answer += code[i]);
    }
  }
  return answer.length > 0 ? answer : "EMPTY";
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
public static String solution(String code) {
        StringBuilder ret = new StringBuilder();
        int mode = 0;
        
        for (int i = 0; i < code.length(); i++) {
            char c = code.charAt(i);
            
            if (c == '1') {
                mode = 1 - mode;
            } else {
                if (mode == 0 && i % 2 == 0) {
                    ret.append(c);
                } else if (mode == 1 && i % 2 == 1) {
                    ret.append(c);
                }
            }
        }
        
        return ret.length() == 0 ? "EMPTY" : ret.toString();
    }
}